docs/auth.md

```diff
@@ -0,0 +1,259 @@
+# Serving Certificates, Authentication, and Authorization
+
+This document outlines how the service catalog API server handles
+authentication and authorization, and the steps needed to set this up in
+your cluster.
+
+Additionally, it provides instructions for generating the serving
+certificates used to securely serve the service catalog API.
+
+## Certificates Overview
```

Comment:
This section is really great and illuminating.

If there is relevance to k8s at large, I would hate for it to be stuck in our repo.